environs/manual/sshprovisioner/sshprovisioner.go

```diff
@@ -41,17 +41,22 @@ import (
 //
 // authorizedKeys may be empty, in which case the file
 // will be created and left empty.
-func InitUbuntuUser(host, login, authorizedKeys string, privateKeys string, read io.Reader, write io.Writer) error {
+func InitUbuntuUser(host, login, authorizedKeys string, privateKeys string, knownHostsFile string, read io.Reader, write io.Writer) error {
 	logger.Infof("initialising %q, user %q", host, login)
 
+	var options ssh.Options
+
+	// known hosts file was set
+	options.SetKnownHostsFile(knownHostsFile)
```

Comment:
We should only call this if we actually have something to set. It is important to keep intent here, even if it doesn't do anything with an empty string.